File: src/main/java/com/smarthome/backend/model/TimeTrigger.java
package com.smarthome.backend.model;

import java.util.List;

import lombok.Data;

/**
 * Konfiguration für einen Zeit-Trigger.
 */
@Data
public class TimeTrigger {
    private String frequency; // 'daily' | 'weekly' | 'monthly' | 'yearly'
    private String time; // Format HH:mm
    private List<Integer> weekdays; // Optional, 0-6 (Sonntag-Samstag) für wöchentlich

    public String getFrequency(){
        return frequency;
    }

    public String getTime(){
        return time;
    }

    public List<Integer> getWeekdays(){
        return weekdays;
    }
}

File: src/main/node/com/smarthome/backend/model/TimeTrigger.ts
/**
 * Konfiguration für zeitbasierte Trigger.
 * Kompatibel mit Frontend-Datenmodell.
 * 
 * @property frequency - Häufigkeit: 'once' | 'daily' | 'weekly' | 'monthly' | 'yearly'
 * @property time - Uhrzeit im Format "HH:mm"
 * @property weekdays - Array von Wochentagen (0=Sonntag, 6=Samstag) für weekly
 * @property dayOfMonth - Tag des Monats (1-31) für monthly
 * @property month - Monat (1-12) für yearly
 * @property dayOfYear - Tag des Jahres für yearly (Alternative zu month+dayOfMonth)
 */
export class TimeTrigger {
  frequency?: string; // 'once' | 'daily' | 'weekly' | 'monthly' | 'yearly'
  time?: string;
  weekdays?: number[];
  dayOfMonth?: number;
  month?: number;
  dayOfYear?: number;

  constructor(init?: Partial<TimeTrigger>) {
    Object.assign(this, init);
  }
}
